From 2c5b0d5d6eba3bda4e83f489da546060fa8f3a2b Mon Sep 17 00:00:00 2001
From: Zhu Zihao <all_but_last@163.com>
Date: Sat, 26 Mar 2022 16:54:43 +0800
Subject: [PATCH] Unbundle robin-map.

This patch makes libobjc2 use robin-map header from system instead of cloning
from submodule.
---
 CMakeLists.txt | 8 --------
 arc.mm         | 2 +-
 2 files changed, 1 insertion(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e2746e9..cf731f1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,14 +86,6 @@ else ()
 	list(APPEND libobjc_C_SRCS eh_personality.c)
 endif (WIN32)
 
-if (NOT EXISTS "${CMAKE_SOURCE_DIR}/third_party/robin-map/include/tsl/robin_map.h")
-	message(FATAL_ERROR "Git submodules not present, please run:\n\n"
-						" $ git submodule init && git submodule update\n\n"
-						"If you did not checkout via git, you will need to"
-						"fetch the submodule's contents from"
-						"https://github.com/Tessil/robin-map/")
-endif ()
-
 # For release builds, we disable spamming the terminal with warnings about
 # selector type mismatches
 if (CMAKE_BUILD_TYPE STREQUAL Release)
diff --git a/arc.mm b/arc.mm
index c96681f..cc93e7f 100644
--- a/arc.mm
+++ b/arc.mm
@@ -3,7 +3,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
-#include "third_party/robin-map/include/tsl/robin_map.h"
+#include <tsl/robin_map.h>
 #import "lock.h"
 #import "objc/runtime.h"
 #import "objc/blocks_runtime.h"
-- 
2.34.0

